FindMe/Parse Manager/DatabaseManager.m

```diff
@@ -104,10 +119,20 @@ + (void)fetchContacts:(void(^)(NSArray *contacts))completion
         }
         else {
             for (PFObject *contactObject in contacts) {
-                Contact *contact = [DatabaseManager getContactFromPFObject:contactObject];
-                [fetchedContacts addObject:contact];
+                dispatch_group_enter(group);
+                [DatabaseManager contactFromPFObject:contactObject withCompletion:^(Contact * _Nonnull contact) {
+                    if (contact) {
+                        [fetchedContacts addObject:contact];
+                        dispatch_group_leave(group);
+                    }
+                    else {
+                        dispatch_group_leave(group);
+                    }
```

Comment:
Can clean up a little and just have 

dispatch_group_leave(group);

after the condition.